src/client/components/movies/Movies.js

```diff
@@ -1,31 +1,67 @@
 import React from 'react';
-import styles from './Movies.module.css';
-import MovieContainer from './Movie/MovieContainer';
+import { getAllMovies } from '@root/client/store/store';
+import Movie from '@root/client/components/movies/Movie/Movie'
+import './Movies.module.css'
+import Preloader from '../shared/preloader/Preloader';
 
+class Movies extends React.Component {
 
-const Movies = (props) => {
+  constructor() {
+    super();
+    this.isUnmount = false;
+    this.isLoading = true;
+    this.state = {
+      movies: [],
+    }
+  }
+
+  componentDidMount() {
+    this.isLoading = true;
+    getAllMovies().then((mov) => {
+      if (this.isUnmount === false) {
+        this.setState({ movies: [...mov] });
+      }
+    });
+    this.isLoading = false;
+  }
 
-  if (props.movies.length === 0) {
-    return (
-      <div className={styles.emptyMovies}>
-        <span>No films found </span>
-      </div>
-    )
+  componentWillUnmount() {
+    this.isUnmount = true;
   }
-  else {
 
-    let moviesList = props.movies.map((movie)=> {
+  render() {
+
+    if (this.state.movies.length === 0) {
```

Comment:
don't do such nested if constructor it is weird instead of this do next
`if (this.isLoading) { return (...) } `
`if (!this.state.movies.length) return (...)`
...